schemas/service.proto

```diff
@@ -33,6 +33,17 @@ message PutWfSpecRequest {
   string entrypoint_thread_name = 6;
   optional WorkflowRetentionPolicy retention_policy = 8;
   reserved 9;
+  optional AllowedUpdateType allowed_updates = 10;
+}
+
+// Defines the allowed update type
+// ALL - Creates either a revision or majorVersion when WfSpec changes
+// MINOR_REVISION_ONLY - Allow revision updates but reject majorVersion
+// NONE - Reject any update
+enum AllowedUpdateType {
```

Comment:
One quirk with protobuf is that we won't be able to have any other enum value in our entire package that is called `ALL`, `MINOR_REVISION_ONLY`, or `NONE`.

I definitely don't think we should reserve `ALL` or `NONE` for this use case. Can we please put a prefix? Or make it `ALL_UPDATES`, `MINOR_REVISION_UPDATES`, and `NO_UPDATES`?

Reply:
Oh understood, will do.